lib/event_icon_builder.rb

```diff
@@ -0,0 +1,36 @@
+class EventIconBuilder
+  attr_reader :events, :start_id, :count
+
+  DEFAULT_COUNT = 50
+
+  def initialize(user_id:, start_id: 0, count: DEFAULT_COUNT)
+    @user_id = user_id
+    @start_id = start_id.to_i
+    @count = define_count(count: count)
+    @events = User.find(user_id).events.limit(count).offset(start_id)
+  end
+
+  def build_payload
+    {
+      start_id: start_id,
+      count: count,
+      events: event_icons
+    }
+  end
+
+  private
+
+  def event_icons
+    events.each_with_index.map do |event, index|
+      # The index + start_id means if you ask for start_id 30, then the first event in 
+      # the returned payload will be 30.  It counts upwards from there to stop at
+      # start_id + count.
+      event.to_icon(event_number: index + start_id)
```

Comment:
Noted on the camel case, no biggie to change to that. And I like your idea about having the plural model (the payload) be composed if singular models.

Also makes it easier to generalize I think.